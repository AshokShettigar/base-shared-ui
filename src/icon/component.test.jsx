import { shallow } from 'enzyme';
import { Icon } from 'platform-common-ui';

describe('icon', function() {
    it('should render the component', function() {
        const wrapper = shallow(<Icon icon="some-icon" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with class name', function() {
        const wrapper = shallow(<Icon className="my-class" icon="some-icon" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should set custom spriteUrl', function() {
        const wrapper = shallow(<Icon spriteUrl="diff_url/sprite.svg" icon="some-icon" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger onClick function', function() {
        const handleClick = jest.fn();
        const wrapper = shallow(<Icon icon="some-icon" onClick={handleClick} />);

        wrapper.find('i').simulate('click');

        expect(handleClick).toHaveBeenCalled();
    });
});
