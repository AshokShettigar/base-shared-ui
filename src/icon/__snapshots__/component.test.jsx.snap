// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon should render the component 1`] = `
<i
  className="cn-icon cn-icon--size-extra-small"
  data-component="Icon"
>
  <svg>
    <use
      xlinkHref="sprites.svg#some-icon"
    />
  </svg>
</i>
`;

exports[`icon should render the component with class name 1`] = `
<i
  className="cn-icon my-class cn-icon--size-extra-small"
  data-component="Icon"
>
  <svg>
    <use
      xlinkHref="sprites.svg#some-icon"
    />
  </svg>
</i>
`;

exports[`icon should set custom spriteUrl 1`] = `
<i
  className="cn-icon cn-icon--size-extra-small"
  data-component="Icon"
>
  <svg>
    <use
      xlinkHref="diff_url/sprite.svg#some-icon"
    />
  </svg>
</i>
`;
